### DISCUSSION TITLE
initrd file size cause problems

### DISCUSSION THREAD

**Discussion by axeprpr** [ipxe/ipxe#1306] [General]
hi, now i try to use ipxe to install proxmox. 
i use https://github.com/morph027/pve-iso-2-pxe
i met a problem that it can not boot when initrd file is too large.
the origin initrd file of proxmox can boot from ipxe.
after we use pve-iso-2-pxe rebuiding the initrd file. it boot failed.
i think this is a problem of ipxe.efi?
another problem is,
the https://github.com/morph027/pve-iso-2-pxe write ipxe file like this:
```
set opts "vga=791 video=vesafb:ywrap,mtrr ramdisk_size=16777216 rw quiet initrd=initrd"
set kernel "${webserver}/linux26 ${opts} splash=verbose proxdebug proxmox-start-auto-installer"
initrd ${webserver}/initrd
chain ${kernel}
```
it boot failed because the spaces parsed to %20.

**Comment by NiKiZe** [comment]
I can not see that you have shown anything that suggests that this is a problem with iPXE. (So moving to discussions)

Do you get any kind of error message? What is your exact output?

How large is the initrd?

And yes, the kernel options should not be part of the URL.

**Reply by axeprpr** [reply]
i use the latest version of ipxe.efi. `wget https://boot.ipxe.org/ipxe.efi`
```
root@debian:/home/netboot/boot/proxmox# du -sh initrd
1.5G    initrd
```
![image](https://github.com/user-attachments/assets/f3b4fa37-e1a9-4e1a-81cd-8e2f7e814f31)

**Reply by NiKiZe** [reply]
Did you follow the URL that is shown?
https://ipxe.org/7f048289
Instead of `chain`, use `kernel` and `boot`, but insert `imgstat` in-between so you can verify that the kernel is verified as a EFI boot target. I'm sure it is, but still good to see the data.

If the kernel isn't signed for secure boot, you need to disable secure boot in firmware settings.

What happens is that the kernel is set for booting, and the firmware is asked to run it but it returns. Not sure if this is your firmware, or if it is the kernel that returns 0x89, but I would assume your firmware.

**Reply by axeprpr** [reply]
i use vmware to test. secure boot is disabled.

i add DEBUG=efi_image and build a new file.
![image](https://github.com/user-attachments/assets/78b66aa1-4862-470f-a4e2-8a70df4bf6e6)

**Reply by NiKiZe** [reply]
So you didn't grab the `imgstat` output as I suggested? Sure, DEBUG is helpful in some cases, but in many others its just in the way... Like now where you are seeing messages which is not related to your actuall problem.

**Reply by axeprpr** [reply]
yes.
![image](https://github.com/user-attachments/assets/c829d750-c5f3-4bdc-a9db-07e1173d57c3)

**Reply by NiKiZe** [reply]
> Not sure if this is your firmware, or if it is the kernel that returns 0x89, but I would assume your firmware.

**Reply by mcb30** [reply]
The underlying error reported by `StartImage()` is `EFI_OUT_OF_RESOURCES`.  The initrd is fairly large (1.6GB), so most likely this is the kernel's EFI stub reporting that it does not have enough memory to load a copy of the initrd image.

Try with a smaller initrd image, or with more RAM.

**Comment by axeprpr** [comment]
```
#!ipxe
set webserver http://192.168.132.130:5000/boot/proxmox
initrd ${webserver}/initrd
kernel ${webserver}/linux26 vga=791 video=vesafb:ywrap,mtrr ramdisk_size=16777216 rw quiet splash=verbose initrd=initrd proxdebug
imgstat
boot || shell
```

**Comment by xkolk** [comment]
It's well known problem with no published workaround, it also affects grub-efi
https://www.ibm.com/support/pages/system-boot-ends-grub-out-memory-oom

**Comment by xkolk** [comment]
I would suggest making initrd/init script a bit smarter and able to fetch the iso after boot
